src/v0/util/salesforce/factory.ts

```diff
@@ -0,0 +1,23 @@
+import Legacy from './legacy';
+import OAuth from './oauth';
+import {
+  LEGACY,
+  OAUTH,
+  SALESFORCE_OAUTH,
+  SALESFORCE_OAUTH_SANDBOX,
+} from '../../destinations/salesforce/config';
+import { ACCESS_TOKEN_CACHE_TTL_SECONDS } from '../../destinations/wootric/config';
+
+const oauth = new OAuth();
+const legacy = new Legacy(ACCESS_TOKEN_CACHE_TTL_SECONDS);
+
+const getAuthInfo = async (event: { destination: { DestinationDefinition: { Name: string } } }) => {
+  const { Name } = event.destination.DestinationDefinition;
+  const lowerCaseName = Name?.toLowerCase?.();
+  if (lowerCaseName === SALESFORCE_OAUTH_SANDBOX || lowerCaseName === SALESFORCE_OAUTH) {
+    return oauth.collectAuthorizationInfo(event);
+  }
+  return legacy.collectAuthorizationInfo(event);
+};
+
+export default { [OAUTH]: oauth, [LEGACY]: legacy, getAuthInfo };
```

Comment:
instead of using OAuth or legacy as keys, should we use destination definition name as key

Reply:
We can do it. But what would be the advantage of doing it ?

Reply:
Anyway moved away from using enums to destination definition name